src/main/java/org/mifos/processor/bulk/format/RestRequestConvertor.java

```diff
@@ -0,0 +1,74 @@
+package org.mifos.processor.bulk.format;
+
+import org.mifos.processor.bulk.schema.BatchRequestDTO;
+import org.mifos.processor.bulk.schema.Party;
+import org.mifos.processor.bulk.schema.PartyType;
+import org.mifos.processor.bulk.schema.Transaction;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+@Component
+public class RestRequestConvertor implements EntityMapper<BatchRequestDTO, Transaction> {
+
+    @Override
+    public BatchRequestDTO convertTo(Transaction object) {
+        Party creditParty = new Party();
+        creditParty.setKey(PartyType.fromValue(object.getPayeeIdentifierType()).getValue());
+        creditParty.setValue(object.getPayeeIdentifier());
+
+        Party debitParty = new Party();
+        debitParty.setKey(PartyType.fromValue(object.getPayerIdentifierType()).getValue());
+        debitParty.setValue(object.getPayerIdentifier());
+
+        // creating the DTO
+        BatchRequestDTO batchRequestDTO = new BatchRequestDTO();
+        batchRequestDTO.setAmount(object.getAmount());
+        batchRequestDTO.setCurrency(object.getCurrency());
+        batchRequestDTO.setCreditParty(List.of(creditParty));
+        batchRequestDTO.setDebitParty(List.of(debitParty));
+        batchRequestDTO.setDescriptionText(object.getNote());
+        batchRequestDTO.setSubType(object.getPaymentMode());
+
+        return batchRequestDTO;
+    }
+
+    @Override
+    public Transaction convertFrom(BatchRequestDTO object) {
```

Comment:
use better name for variables

Reply:
please suggest a better name, since its implementing a generic interface so its named as to and from. I am not able to think of any better name.